src/jdbc/resultset.clj

```diff
@@ -52,9 +58,12 @@
                (when (.next rs)
                  (cons (values) (lazy-seq (thisfn)))))
         records (fn thisfn []
-                  (when (.next rs)
-                    (-> (zipmap keyseq (values))
-                        (cons (lazy-seq (thisfn))))))]
+                  (let [keyseq (if metadata->identifiers
+                                 (metadata->identifiers metadata)
+                                 (mapv (comp keyword identifiers (fn [^long i] (.getColumnLabel metadata i))) idseq))]
+                    (when (.next rs)
```

Comment:
Ah true, I didn't think of that, I was just trying to easily avoid computing it when as-rows? was true.

Basically I just want access to all the meta data to compute the identifiers, that way I can automatically add ? characters after bools, and possibly do something with Enums and DATERANGEs in postgres.

I didn't want to change existing behaviour which is why I just added another optional parameter.